invenio/utils/deprecation.py

```diff
@@ -87,17 +89,49 @@ class RemovedInInvenio22Warning(DeprecationWarning):
     """Mark feature that will be removed in Invenio version 2.2."""
 
 
+# Modified version of http://stackoverflow.com/a/22376126/1727265
+def warn_with_traceback(message, category, filename, lineno, file=None,
+                        line=None):
+    """Log deprecation warnings to debug logger."""
+    # Format
+    message.args += 'Full backtrace in the logfile if DEBUG is enabled',
+    formatted_warning = warnings.formatwarning(message, category, filename,
+                                               lineno, line)
+
+    # Warn
+    log = file if hasattr(file, 'write') else sys.stderr
```

Comment:
IMHO we should check that `if file is not None` otherwise "`if not hasattr(file, 'write')`" indicates runtime error.